### PR TITLE
fix(core): remove list focus trap

### LINKED ISSUE #8687: List items traps focus
#### Is this a bug, enhancement, or feature request?
bug

#### Briefly describe your proposal.
At the moment the list traps the focus and the user cannot go out with the keyboard.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
latest

### PR DESCRIPTION
**Pull request by shrvr**
## Related Issue(s)

closes #8687 

## Description

Removed list focus trap on tab key.

### CODE REVIEW

**Review by mikerodonnell89 — APPROVED** [review]
This works well and addresses the issue with FocusKeyManager losing sight of its active item when tab/shift+tab is used

**Review by platon-rov — CHANGES_REQUESTED** [review]
Before the action list fix pr (https://github.com/SAP/fundamental-ngx/pull/8458) we had only the first item in list focusable with tab and nov every item has tab stop.

**Comment by shrvr** [comment]
> Before the action list fix pr (#8458) we had only the first item in list focusable with tab and nov every item has tab stop.

So to Navigate within the list,  should only arrow keys be used? Because for action sheet the navigation is by both Tab and arrow keys, and both share Keyboard Service.
Or I can add one more argument to `setKeyboardService ` to enable or disable tab key listening.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
I think refactoring the onKeyDown function of `keyboard-support.service.ts` to this should do it: 

```
    /** @hidden */
    onKeyDown(event: KeyboardEvent): void {
        this._keyManager.onKeydown(event);
        if (KeyUtil.isKeyCode(event, TAB) && this._focusTrapOnTab) {
            event.preventDefault();
            hasModifierKey(event, 'shiftKey') ? this.keyManager.setPreviousItemActive() : this.keyManager.setNextItemActive();
        }
    }
```

File: libs/core/src/lib/utils/services/keyboard-support/keyboard-support.service.ts [hunk 1]
```diff
@@ -38,13 +52,19 @@ export class KeyboardSupportService<T> {
     onKeyDown(event: KeyboardEvent): void {
```
Comment (mikerodonnell89) [inline libs/core/src/lib/utils/services/keyboard-support/keyboard-support.service.ts:52]: I think try refactoring this function to 

```suggestion
    /** @hidden */
    onKeyDown(event: KeyboardEvent): void {
        this._keyManager.onKeydown(event);
        if (KeyUtil.isKeyCode(event, TAB) && this._focusTrapOnTab) {
            event.preventDefault();
            hasModifierKey(event, 'shiftKey') ? this.keyManager.setPreviousItemActive() : this.keyManager.setNextItemActive();
        }
    }
```

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Pull request merged**